### PR TITLE
feat(config): optional IPv4-only connection mode (#606)

### LINKED ISSUE #606: Connection error "DNS server returned answer with no data" — root cause: strict AF_UNSPEC resolution vs Incapsula CDN with no AAAA record
## Describe the bug

The integration repeatedly fails to refresh with:

```
Refresh failed: Connection error with URL https://customers.securitasdirect.fr/owa-api/graphql: [Errno None] DNS server returned answer with no data
```

## Diagnosis

Manual DNS resolution works fine and shows the actual root cause:

- `customers.securitasdirect.fr` is a CNAME to `poay6y8.impervadns.net` (Incapsula/Imperva WAF).
- The AAAA query for that name returns a properly signed NODATA response (SOA from `impervadns.net`, no server error) — Incapsula simply does not publish an IPv6 record for this endpoint.
- The A query succeeds and returns a valid IP (`45.60.154.141` at time of testing), but with a **30-second TTL** — this IP rotates frequently, so any static workaround (hosts file, static DNS entry) is short-lived and not a viable fix.
- The actual failure comes from the combined A+AAAA (`AF_UNSPEC`) resolution used by `aiohttp` via `aiodns`/`pycares`: when one of the two queries fails or returns NODATA, the whole resolution call fails instead of falling back to the successful A record. This matches a known and currently-discussed behavior in DNS resolver implementations under strict RFC 8020 interpretation (see e.g. the musl libc mailing list discussion from August 2026 on the same failure mode — Alpine/musl-based images are affected the same way).

## Environment

- Home Assistant Core: 2026.9.2
- Integration version (HACS): 5.7.0 ("Verisure OWA" / securitas)
- aiohttp: 3.14.3
- aiodns: 4.0.4
- pycares: 5.0.1
- Installation type: Home Assistant OS

## Suggested fix

Force `family=socket.AF_INET` for HTTP requests made by this integration to `customers.securitasdirect.fr` (or configure the underlying aiohttp client/connector to avoid dual A+AAAA resolution), since the target CDN endpoint has no IPv6 support and its A record has a very short (30s) TTL that rules out any DNS-level workaround.

---

*Note: the diagnosis above and the suggested fix were worked out with the help of an AI assistant (Claude) — I'm not a networking/DNS specialist myself, so please treat the suggested fix as a lead to investigate rather than a verified patch, and let me know if you need more logs or tests from my setup.*

### PR DESCRIPTION
**Pull request by clintongormley**
## What this fixes

Fixes #606. On some networks the integration repeatedly fails to refresh and the alarm drops to *unavailable* with:

```
Refresh failed: Connection error with URL https://customers.securitasdirect.fr/owa-api/graphql: [Errno None] DNS server returned answer with no data
```

The Verisure endpoint (behind an Incapsula/Imperva CDN) publishes **no IPv6 (AAAA) record**. By default the HTTP stack resolves IPv4 and IPv6 together; when the IPv6 half comes back empty, some resolvers treat the whole lookup as failed instead of using the IPv4 address that resolved fine. The result is a persistent failure that a hosts-file workaround can't fix (the A record has a 30-second, rotating TTL).

## What this adds

A new **"Force IPv4-only connections"** toggle in the post-setup **Configure → Advanced** options, **off by default**. When enabled, the integration talks to Verisure over IPv4 only, so it never issues the IPv6 query that was failing. Because the endpoint has no IPv6 address, nothing is lost by turning it on.

Implementation notes for reviewers:

- When on, `_build_http_session` builds a dedicated client with Home Assistant's own `async_create_clientsession(hass, family=socket.AF_INET, auto_cleanup=False)`, so it keeps HA's SSRF-redirect middleware, pooled connector and managed SSL context — only the resolver's address family changes.
- `auto_cleanup=False` because the client is shared per-account (by username) across config entries; the integration owns its lifecycle and **detaches** it (never `close()`, which would tear down HA's shared connector) when the last entry using it unloads, and also on a login failure before the session is registered, so setup retries don't leak clients.
- The toggle is **not** shown during initial setup, where the login has already run on the shared client and the toggle would have no effect.
- **Known limitation:** two config entries for the same account share one client whose address family is fixed by whichever entry created it first, so the toggle only fully takes effect once every entry for that account has reloaded. This is noted in a code comment and the changelog.

## Docs & i18n

- New label and help text translated into all seven locales.
- README troubleshooting section explaining the symptom and when to try the toggle.
- `docs/architecture.md` updated (the options-flow Advanced section now differs from initial setup).
- `CHANGES.md` entry under v5.9.0, crediting @Teuqol for the diagnosis.

## Tests

- New `tests/test_force_ipv4.py`: the off/on session choice, the IPv4-only connector family, release-on-last-reference, detach-on-unload, and detach-on-failed-login (the last two guarding against client leaks).
- Added `tests/test_config_flow.py` cases: the toggle appears in the options Advanced section (and is pre-filled), and is omitted from initial setup.

The reporter treated their AI-assisted diagnosis as a lead; I verified the mechanism against the code and covered it with tests, but note I could not reproduce a live broken-IPv6 endpoint — the fix rests on IPv4-only resolution skipping the AAAA query, which is well-established behaviour.

### CODE REVIEW

File: custom_components/securitas/__init__.py [hunk 1]
```diff
@@ -591,9 +640,29 @@ async def _get_or_create_session(
...
+                # see ``_build_http_session``). Detach is a no-op safeguard for
+                # the borrowed shared client (owned is False there).
+                if owned:
+                    http_session.detach()
+                raise
+            sessions[username] = {
+                "hub": client,
+                "ref_count": 1,
+                # Present only when we forced IPv4 and built our own client:
+                # it is detached on unload so reloads don't leak clients. The
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply custom_components/securitas/__init__.py:662]: Good catch — confirmed and fixed. HA doesn't call `async_unload_entry` on `ConfigEntryNotReady`, so a post-login setup failure left this entry's session reference (and, for a forced-IPv4 account, the owned client) registered; a retry then took the reuse branch and inflated `ref_count`, so it never returned to zero to detach the client.

Added `_release_session_reference()` and call it from the `_fetch_and_cache_installations` failure path (the one post-login I/O call that raises `ConfigEntryNotReady` — coordinators are constructed without network calls, and `get_services` is already caught), so the retry starts from a clean session. Covered by a new test (`test_fetch_failure_detaches_owned_session`) that drives `async_setup_entry` with the fetch failing and asserts the client is detached and the session removed.

**Comment by clintongormley** [comment]
Acted on the four findings the review left unposted (they appear only in the review body, so there was no thread to answer on). All four were checked against the branch head first and all four were still live. Fixed in 3747a35.

**Signing in again ignored the setting** (`config_flow.py` reauth) — confirmed. `_create_client()` always built the hub on `async_get_clientsession`, the dual-stack client, and the reauth step went straight through it. That is the exact lookup the option exists to skip, and a refresh failure is what pushes an affected user into reauth, so the one sign-in that had to work was the one that could not. `_create_client(force_ipv4=...)` now routes through the same `_build_http_session` helper setup uses, and the reauth step passes the entry's saved value with the same precedence `_build_config_dict` applies (options, then data, then the default). A client the flow builds for itself is detached in `async_remove` — Home Assistant calls that whenever a flow is removed, on success or abort — and again at the top of `_create_client`, so a retried step cannot stack clients up. Three tests in `TestReauthHonoursForceIpv4`; two of them were watched failing first, and hard-coding `force_ipv4=False` back into the reauth step makes both fail again, so they discriminate.

**No coverage of the option reaching setup** — confirmed. The two ends were tested but not the join. `TestSavedOptionReachesTheConnection` now drives `async_setup_entry` with the option saved on, saved off and never saved, and asserts which client the hub was handed. These passed on arrival, since the code was already correct, so they were verified by mutation instead: making `_build_config_dict` ignore the option fails the on-case, and hard-coding it to `True` fails both off-cases.

**README overstated how fast it takes effect** — confirmed, and it contradicted the known limitation in the PR description. "It applies straight away" is replaced with a paragraph explaining that entries sharing a username share one connection whose address family is fixed by whichever entry loads first, so the setting should be turned on for every entry using that account, followed by a restart. The same instruction, one sentence, was added to the help text beside the tick box in `strings.json` and all seven locales.

**pt-BR wording** — confirmed. Every sibling locale uses the past participle (`fallida`, `fallita`, `en échec`) and European Portuguese already said `falhada`, so pt-BR now says `falhada` too. That keeps it consistent with its own sibling rather than restructuring the clause.

Also updated `CHANGES.md` and `docs/architecture.md`. One side effect worth flagging: `config_flow` no longer imports `async_get_clientsession`, since the single dispatch point is now `_build_http_session`, so `PATCH_SESSION` in `tests/test_config_flow.py` was repointed at `custom_components.securitas.async_get_clientsession`. Full suites are green (842 unit, 1073 integration), as are ruff, pyright, and the translation and docs drift checks.

File: tests/test_force_ipv4.py [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+to the working IPv4 answer. The ``force_ipv4`` option lets an affected user
+switch this integration's requests to an IPv4-only lookup, which never asks
+the IPv6 question in the first place.
+"""
+
+import socket
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from homeassistant.exceptions import ConfigEntryNotReady
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply tests/test_force_ipv4.py:15]: This one is the wrong way round, so I've made the opposite edit rather than the suggested one.

The module-level import on line 15 is used: `TestSavedOptionReachesTheConnection._session_setup_built` calls `pytest.raises(ConfigEntryNotReady)` and imports nothing locally, so removing line 15 would break that test with a `NameError`. Only the older `test_fetch_failure_detaches_owned_session` has a local copy, and a local import shadows inside that one function — it can't satisfy the other test.

The stated consequence is also checkable, and it doesn't hold: `ruff check --select F401 tests/test_force_ipv4.py` passes locally, and the **Ruff lint & format** job passed on 619c7ec, the exact commit reviewed. All 14 checks passed on it.

There is a real tidy-up here though, in the other direction: the *local* import is the redundant one now that the module-level import exists. Removed in 570d369, which also takes away the shadowing that probably prompted this. Ruff and the suite are green after it.

**Comment by clintongormley** [comment]
Closing in favour of #614, which fixes the same issue (#606) a better way.

Review of this branch turned up two things that undermined the design rather than just needing patches:

**The people who needed the toggle most could not reach it.** Setup signs in at the first screen, and the toggle only existed on the options screen that appears once an entry exists. Anyone installing fresh on an affected network failed at screen one and never got to the setting the README told them to use.

**The hand-managed HTTP session was unnecessary.** Home Assistant's `async_get_clientsession(hass, family=...)` already caches one session — and one connector — per address family, with its own shutdown cleanup. It is present in HA 2025.2.0, this integration's declared minimum, and it goes through the same builder as the default client, so it carries the same SSRF middleware, SSL context and shared resolver. All the ownership bookkeeping on this branch was guarding a thin wrapper around a connector Home Assistant was already sharing — and every bug found here lived in one of its release paths.

Checking the premise across all ten supported countries also showed the problem is not French-specific: none of the endpoints has an IPv6 address, they are all CNAMEs into the same Imperva edge, and they all resolve to one IPv4 address. So the IPv6 half of the lookup can only ever come back empty for everyone, which makes asking for IPv4 the right default rather than an opt-in.

#614 therefore always connects over IPv4 and falls back to the ordinary lookup only when the attempt provably never reached the server — which keeps IPv6-only hosts using NAT64 working, without anyone needing to know which network they are on. A timeout deliberately does not trigger the fallback, since this integration does not resend a sign-in to a busy server.

Net: −298 lines against this branch, no new setting, nothing to translate, and both of the bugs still open here gone by construction.

Thanks to @Teuqol for the diagnosis that found the cause — it carries over to #614.
